### PR TITLE
support training from .raw.png

### PR DESCRIPTION
**Pull request by bertsky**
obviously depends on https://github.com/tesseract-ocr/tesseract/pull/3962 as well

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -254,6 +262,10 @@ $(ALL_LSTMF): $(ALL_FILES:%.gt.txt=%.lstmf)
 	set -x; \
 	tesseract "$<" $* --psm $(PSM) lstm.train
 
+# do not search for implicit rules here:                                                                                                     
+%.png: ;
+%.tif: ;
+%.gt.txt: ;
```
Comment (stweil) [inline Makefile:268]: That's an unrelated modification. I'd prefer a separate pull request for it.

File: Makefile [hunk 2]
```diff
@@ -246,6 +250,10 @@ $(ALL_LSTMF): $(ALL_FILES:%.gt.txt=%.lstmf)
 	set -x; \
 	tesseract "$<" $* --psm $(PSM) lstm.train
 
+%.lstmf: %.raw.png %.box
+	set -x; \
+	tesseract "$<" $* --psm $(PSM) lstm.train
+
```
Comment (stweil) [inline Makefile:256]: Please move that down some lines to keep the alphabetic order.

File: Makefile [hunk 3]
```diff
@@ -226,6 +226,9 @@ $(ALL_GT): $(ALL_FILES) | $(OUTPUT_DIR)
 %.box: %.bin.png %.gt.txt
 	PYTHONIOENCODING=utf-8 python3 $(GENERATE_BOX_SCRIPT) -i "$*.bin.png" -t "$*.gt.txt" > "$@"
 
+%.box: %.raw.png %.gt.txt
+	PYTHONIOENCODING=utf-8 python3 $(GENERATE_BOX_SCRIPT) -i "$*.raw.png" -t "$*.gt.txt" > "$@"
+
```
Comment (stweil) [inline Makefile:231]: Please move that down some lines to keep the alphabetic order.

File: Makefile [hunk 4]
```diff
@@ -238,6 +241,7 @@ $(ALL_LSTMF): $(ALL_FILES:%.gt.txt=%.lstmf)
 	$(file >$@) $(foreach F,$^,$(file >>$@,$F))
 	python3 shuffle.py $(RANDOM_SEED) "$@"
 
+.PRECIOUS: %.lstmf
```
Comment (stweil) [inline Makefile:244]: That's an unrelated modification. I'd prefer a separate pull request for it.

**Review by stweil — CHANGES_REQUESTED** [review]

**Review by stweil — COMMENTED** [review]
Thanks.

**Comment by stweil** [comment]
Do you want to fix your git `user.email` setting for future contributions?

**Comment by bertsky** [comment]
> Do you want to fix your git `user.email` setting for future contributions?

Why, what's wrong with it? (For this one, I've used GH directly.)

**Comment by stweil** [comment]
GitHub does not add your normal e-mail address as author (unless you have configured it there).
